internal/nginx/main.go

```diff
@@ -47,11 +50,16 @@ var StreamSocket = "/tmp/ingress-stream.sock"
 
 var statusLocation = "nginx-status"
 
-var socketClient = buildUnixSocketClient()
-
 // NewGetStatusRequest creates a new GET request to the internal NGINX status server
-func NewGetStatusRequest(path string) (int, []byte, error) {
+func NewGetStatusRequest(path string, timeouts ...time.Duration) (int, []byte, error) {
```

Comment:
why more than one timeout?

Reply:
because the method `NewGetStatusRequest` has been used in many places,  like [here] (https://github.com/kubernetes/ingress-nginx/blob/7b737df49a5031964f35e51f4e9e8304ddd43497/cmd/dbg/main.go#L118), and it is NOT the health check related calls, so I make timeouts as an optional input, if it's NOT specified, we use default timeout 10s.

Reply:
I prefer not adding this timeout and use the `nginx.HealthCheckTimeout` variable

Reply:
Done!